tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py

```diff
@@ -29,36 +29,48 @@ def __init__(self, marionette):
         Wait(self.marionette).until(lambda m: view.location['y'] == 0)
 
     def tap_wallpaper(self):
+        Wait(self.marionette).until(
+            expected.element_displayed(*self._actions_menu_locator))
         self.marionette.find_element(*self._wallpaper_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
         from gaiatest.apps.wallpaper.app import Wallpaper
         wallpaper = Wallpaper(self.marionette)
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == wallpaper.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == wallpaper.name)
         self.apps.switch_to_displayed_app()
         return wallpaper
 
     def tap_gallery(self):
         self.marionette.find_element(*self._gallery_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
+        Wait(self.marionette).until(
+            expected.element_not_displayed(actions_menu))
```

Comment:
This still doesn't find the element or wait for it to be visible before we tap it. If the element is never visible then this will still pass, even though our expectation is for it to transition from visible to not hidden.